Emails Dashboard: Prevent same forwarder from being added twice

xavier-lc opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Fixes DOTMSD-936

## Proposed Changes

* There was already a check for duplicate forward addresses, but it wasn't taking into account the form's untokenized input (that is, text that's not been fed to the `DataForm` by pressing the `comma` or `Enter` keys). This PR fixes that by taking it into account.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Users shouldn't be able to set the same forwarder more than once.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the live preview
* Go to `/v2/emails`
* Note one of the forwarders you have already configured
* Click on `Add email forwarder`
* Fill the email address, domain, and forward to fields without pressing the `comma` or `Enter` keys
* Check that you can't click on `Save` because it's disabled and you see an appropriate error message:

<img width="1525" height="1167" alt="image" src="https://github.com/user-attachments/assets/099e57ac-38e7-44bb-8faf-40ad82430148" />

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Extract `useUserEmailAddresses`

Commit:
Revert "Extract `useUserEmailAddresses`"

This reverts commit 88e6ae51b27bf37e6fc53b2d43a823ad2e6dbb17.

Commit:
Consider untokenized input when checking for duplicate forwardings

Commit:
Remove unnecessary operator now that we have the `forwardingAddresses` variable

Commit:
Improve check now that we have the `forwardingAddresses` variable

Commit:
Improve msg now that we have the `forwardingAddresses` variable

rcrdortiz commented:
I can still add forwarders to redirects that already exist. Once I add a new forwarder, I can't add it again, but for existing ones, I'm still able to add the same one once.

Maybe we could re-check when clicking the save button, too, in case we have an inconsistent state at some point of the flow.

https://github.com/user-attachments/assets/5d19485c-1b96-4753-a6db-1ac544086873

Commit:
Fix duplicate forwards detection

By using a `Map` to store the forwards, we were only considering one forward per mailbox.

Commit:
Make more readable

Commit:
Keep the `Map` approach

Commit:
Space

rcrdortiz reviewed (approved):
This is working as expected for me now.
